src/utility.ts

```diff
@@ -8,14 +8,20 @@ import { inflate, deflate } from 'pako';
 
 const BASE64_STR = 'base64';
 
+export class WrappedError extends Error {
+  constructor(message: string, public readonly originalErr: unknown) {
```

Comment:
Why not simply use the cause property of Error, i.e. `new Error(message, { cause })` instead of that custom WrappedError with custom properties ?

ref:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#parameters
- https://nodejs.org/docs/latest-v16.x/api/errors.html#new-errormessage-options

Reply:
I would prefer the native approach, but the `cause` option was only added to Node in 16.19.0 and I _think_ this lib targets as far back as 12.x.

Reply:
That could be challenges IMHO (I've just passed actions to node 16 because GitHub is deprecating everything older than node 16) but could you use the same terminology at least ?